File: DSA/Basics/palindrome.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

bool reverseDigits(int n){
    int dup = n;
    int rev= 0;
    int N = n;

    while(N!=0){

        int digit= N%10;
        rev= rev * 10 + digit;
        N=N/10;
    }
    if(dup == rev){
            return true;
        }
    else{
            return false;
        }
}

int main(){
    int n;
    cout << "enter a number to be check: ";
    cin >> n;
    bool revdigits= reverseDigits(n);
    cout<< revdigits << endl;
    return 0;
}

// 345
// %10 5 
// %

File: DSA_python/Basic/palindrome.py
def palindrome(y):
    N = y
    dup = y
    rev = 0

    while N != 0 :
        digit = N %10
        rev = rev * 10 + digit
        N = N //10

    if dup == rev:
        return True
    else:
        return False
    
x= int(input("enter a number: "))
print(palindrome(x))
       


